srv/salt/state/miniconda.sls

```diff
@@ -1,26 +1,36 @@
 # File:/srv/salt/miniconda.sls
 
-download_miniconda_installer:
+check_conda_install:
+  cmd.run:
+    - name: conda --version
+    - runas: root
+    - shell: /bin/bash
+    - env:
+      - PATH: /root/miniconda3/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/root/bin
+
+ensure_miniconda_installer:
   file.managed:
     - name: /home/centos/Miniconda3-latest-Linux-x86_64.sh
     - source: https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
     - skip_verify: True
-    - keep_source: True  # keep to avoid re-downloading.
+    - keep_source: True  # Keep to avoid re-downloading.
     - mode: 700
+    - onfail:
+      - cmd: check_conda_install
+
+install_miniconda:
+  cmd.run:
+    - name: 'sudo /home/centos/Miniconda3-latest-Linux-x86_64.sh -b -u'
+    - onfail:
+      - cmd: check_conda_install
```

Comment:
should be requiring ensure_minicond_installer instead in case the download fails.